Day11/Monkey.cs

```diff
@@ -0,0 +1,90 @@
+﻿internal class Monkey
+{
+    public int MonkeyID { get; init; }
+    private readonly Queue<Item> items;
+    private Action<Item> operation;
+    private readonly int value;
+    private readonly int divider;
+    private readonly int trueTarget;
+    private readonly int falseTarget;
+    private readonly int interestDrop = 3;
+    public long Activeness { get; private set; } = 0L;
+
+    public Monkey(int monkeyID, Queue<Item> items, int divider, int trueTarget, int falseTarget)
+    {
+        MonkeyID = monkeyID;
+        this.items = items;
+        this.divider = divider;
+        this.trueTarget = trueTarget;
+        this.falseTarget = falseTarget;
+        SetOperationToSquare();
+    }
+
+    public Monkey(int monkeyID, Queue<Item> items, string operation, int value, int divider, int trueTarget, int falseTarget)
+    {
+        MonkeyID = monkeyID;
+        this.items = items;
+        this.value = value;
+        this.divider = divider;
+        this.trueTarget = trueTarget;
+        this.falseTarget = falseTarget;
+        SetOperation(operation);
+    }
+
+    public Queue<(int target, Item item)> ProcessAllItems()
+    {
+        var thrownItems = new Queue<(int target, Item item)>();
+
+        while (items.Count != 0)
+        {
+            var item = items.Dequeue();
+            thrownItems.Enqueue(ProcessItem(item));
+        }
+
+        return thrownItems;
+    }
+    public void CatchItem(Item item)
+        => items.Enqueue(item);
+
+    private (int target, Item item) ProcessItem(Item item)
+    {
+        Activeness++;
+        operation(item);
+        DoInterestCorrection(item);
+        return ThrowItem(item);
+    }
+
+    private (int target, Item item) ThrowItem(Item item)
+        => item.Value % divider == 0 ? (trueTarget, item) : (falseTarget, item);
+
+
+    private void DoInterestCorrection(Item item)
+    {
+        if (item.AmICalming)
+        {
+            item.Divide(interestDrop);
+        }
+    }
```

Comment:
Ez kicsit fura hogy a Part 1 calmingot a monkey csinalja de a part2-t a father time

Reply:
illetve a calming property nekem nem illik az item-be, nem igazan az item tulajdonsaga az. vagy a monkey, vagy a father time.
A te megoldasod akkor lennehasznos, ha minden masodik item calming lenne, de a tobbi nem. 
Ha amugy is a modulot a father time-ban csinalod, akkor lehetne ez is ott es akkor a father time donti el es intezi mindkettot es se a monkey se az item nem tud a calming jelensegrol